src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -476,6 +476,9 @@ private void updateScore(Result result, Set<ProgrammingExerciseTestCase> success
         }
         else {
             double weightSum = allTests.stream().filter(testCase -> !testCase.isInvisible()).mapToDouble(ProgrammingExerciseTestCase::getWeight).sum();
+            if (weightSum == 0) {
```

Comment:
Maybe (weightSum == 0) should have the same behavior as (hasDuplicateTestCases || successfulTestCases.isEmpty()) ?

So maybe invoke result.setScore(0); ?

Reply:
Yeah I wasn't really sure but I think your suggestion makes more sense

Reply:
I was not sure as well :)